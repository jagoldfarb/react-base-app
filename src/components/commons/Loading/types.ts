export interface ILoadingProps {
  message?: string;
  size?: number;
}
