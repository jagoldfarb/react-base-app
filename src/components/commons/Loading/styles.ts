import styled from 'styled-components';

import { COLORS } from 'helpers/enums/colors';

export const LoadingContainer = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-top: 20px;
  margin-bottom: 20px;
`;

export const LoadingText = styled.span`
  font-size: 20px;
  color: ${COLORS.gray};
  margin-top: 20px;
`;
