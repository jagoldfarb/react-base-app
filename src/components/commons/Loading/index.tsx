import React, { FunctionComponent } from 'react';

import { CircularProgress } from '@material-ui/core';

import { ILoadingProps } from './types';
import { LoadingContainer, LoadingText } from './styles';

const Loading: FunctionComponent<ILoadingProps> = (props: ILoadingProps) => {
  const { message = '', size = 65 } = props;
  return (
    <LoadingContainer>
      <CircularProgress data-testid="loader" size={size} />
      {!!message && <LoadingText>{message}</LoadingText>}
    </LoadingContainer>
  );
};

export default Loading;
