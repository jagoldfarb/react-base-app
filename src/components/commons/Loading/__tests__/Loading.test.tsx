import React from 'react';
import { render } from '@testing-library/react';

import Loading from '..';

describe('Loading Component', () => {
  it('Should render the loader', () => {
    const { asFragment, getByTestId } = render(<Loading />);
    expect(getByTestId('loader')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should render the loader with text', () => {
    const props = { message: 'Loading...' };
    const { asFragment, getByTestId, getByText } = render(<Loading {...props} />);
    expect(getByTestId('loader')).toBeDefined();
    expect(getByText('Loading...')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
