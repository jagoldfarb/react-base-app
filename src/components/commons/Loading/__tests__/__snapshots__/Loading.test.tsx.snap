// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component Should render the loader 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ cyXInH"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      data-testid="loader"
      role="progressbar"
      style="width: 65px; height: 65px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Loading Component Should render the loader with text 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ cyXInH"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      data-testid="loader"
      role="progressbar"
      style="width: 65px; height: 65px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
    <span
      class="sc-gsTCUz ISRsW"
    >
      Loading...
    </span>
  </div>
</DocumentFragment>
`;
